packages/core/src/utils/htmlValidationUtil.ts

```diff
@@ -0,0 +1,73 @@
+// eslint-disable-next-line max-classes-per-file
+import * as cheerio from 'cheerio';
+import * as logger from './logger';
+
+const rules: Rule[] = [];
+
+export function validateHtmlWithRules(content: string, path: string): void {
+  const $ = cheerio.load(content);
+  rules.forEach((rule) => {
+    rule.validateAndLog($, path);
+  });
+}
+
+class Rule {
+  description: string = '';
+  causeHydration: boolean = false;
```

Comment:
What does `causeHydration` mean here? The rule causes hydration?

Reply:
Resolve as no longer relevant